src/randoop/types/TypeNames.java

```diff
@@ -0,0 +1,50 @@
+package randoop.types;
+
+import randoop.util.PrimitiveTypes;
+
+import plume.UtilMDE;
+
+public class TypeNames {
+  
+  /**
+   * Generates a name for the given type in a format that can be compiled.
+   * 
+   * @param cls is the type for which name is to be generated.
+   * @return string that is a compilable version of type name.
+   */
+  public static String getCompilableName(Class<?> cls) {
```

Comment:
This does not return a result in compilable form.  It looks like it returns it in some variant of JVML form, but I'm not sure what format it actually returns.  Please use a precise term such as "field descriptor", "binary name", "fully-qualified name", etc. rather than JVML format or compilable form.  A brief description of the four (!) formats defined by Java appears at http://types.cs.washington.edu/checker-framework/current/checker-framework-manual.html#signature-annotations .
Also, is there a need to re-implement what exists in other packages?  For example, could you use plume-lib's `fieldDescriptorToBinaryName` (if that is what this is doing, but I now think it's not)?  Or should we add more methods that do the desired conversion, to some library?  (Maybe plume-lib should have a comprehensive ClassNames class that defines all possible conversions, or maybe some other library already does.)